src/InboundServer/index.js

```diff
@@ -172,7 +172,7 @@ class InboundServer {
             const keyName = path.basename(filename, '.pem');
             const keyPath = path.join(this._conf.jwsVerificationKeysDirectory, filename);
             if (eventType === FS_EVENT_TYPES.RENAME) {
-                if (this._jwsVerificationKeys[keyName] == null) {
+                if (fs.existsSync(keyPath)) {
```

Comment:
Please be absolutely sure about when FS_EVENT_TYPES.RENAME events will occur. We do not want to hit the filesystem unnecessarily....this guard condition makes sure we only reload a key if we havent previously loaded it.

Reply:
The original line of code : "if (this._jwsVerificationKeys[keyName] == null) {)" results in a failure of the following, existing, unit test :
 
<img width="1221" alt="Screenshot 2020-04-03 at 16 11 50" src="https://user-images.githubusercontent.com/13884584/78369713-f05f3580-75c5-11ea-9401-a6eaf075b841.png">

Reply:
It seems that if a file exists and then gets renamed/modified the __validationKeys are deleted and then becomes undefined in the test at line 255 because of the  :  delete this._jwsVerificationKeys[keyName]; line 179 - src/inboundServer/index.js.